libs/types-common/src/Data/Misc.hs

```diff
@@ -51,6 +51,9 @@ import Data.Text.Encoding (decodeUtf8, encodeUtf8)
 import Data.ByteString.Lazy (toStrict)
 import Database.CQL.Protocol hiding (unpack)
 #endif
+#ifdef WITH_ARBITRARY
```

Comment:
Out of curiousity; what do we use this flag for?

Reply:
if we don't want to depend on quickcheck because we're building a production executable, we disable it.  in the test suite of a package using this library, we enable it.

Reply:
Right; but rather than cluttering things with CPP would it be possible to do this with Orphan instances? Not saying it's better, but just curious why we took this path instead.

Reply:
if we define orphans in the test suites, we need to define them once in every test suite.  if we define them here, where they live, we can reuse the instances.

there is an alternative approach to CPP involving cabal flags that disable / enable entire modules.  i was thinking of advocating that, but *so much else to do!!*  :)